src/geo/transform.js

```diff
@@ -592,6 +592,38 @@ class Transform {
         const topPoint = vec4.transformMat4(p, p, this.pixelMatrix);
         return topPoint[3] / this.cameraToCenterDistance;
     }
+
+    getCameraPoint() {
+        const pitch = this._pitch;
+        const latOffset = Math.tan(pitch) * (this.cameraToCenterDistance || 1);
+        return this.centerPoint.add(new Point(0, latOffset));
+    }
+
+    getCameraQueryGeometry(queryGeometry: Array<Point>): Array<Point> {
+        const c = this.getCameraPoint();
+
+        if (queryGeometry.length === 1) {
+            return [queryGeometry[0], c];
+        } else {
+            let minX = c.x;
+            let minY = c.y;
+            let maxX = c.x;
+            let maxY = c.y;
+            for (const p of queryGeometry) {
+                minX = Math.min(minX, p.x);
+                minY = Math.min(minY, p.y);
+                maxX = Math.max(maxX, p.x);
+                maxY = Math.max(maxY, p.y);
+            }
+            return [
+                new Point(minX, minY),
+                new Point(maxX, minY),
+                new Point(maxX, maxY),
+                new Point(minX, maxY),
+                new Point(minX, minY)
```

Comment:
This geometry is a polygon (which geojson requires to be closed) while the first one is a line. After looking closely at the intersection tests I think it could be ok to drop this last point but I think being consistent with geojson makes sense.